src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Date.cs

```diff
@@ -61,7 +61,8 @@ public static void TrimDateTimeOffset(Span<byte> buffer, out int bytesWritten)
                 int fractionEnd = 19 + numFractionDigits;
 
                 // Write fraction
-                for (int i = fractionEnd; i >= curIndex; i--)
+                // Leading zeros are written because the fraction becomes zero when it's their turn
+                for (int i = fractionEnd; i > curIndex; i--)
```

Comment:
Can you please explain what test case fails due to this off-by-one issue and how its now fixed? I don't follow how this issue is specific to leading zeros.

Reply:
When we allow the loop to get to the period's index, we overwrite it with a '0'. 

The leading zero comment isn't related to this fix. I Just thought I should add a note to explain how they are written accurately.